packages/cli/src/runner/file-command-runner.ts

```diff
@@ -2,36 +2,95 @@
 // Licensed under the MIT License.
 
 import { Spinner } from 'cli-spinner';
+import * as fs from 'fs';
 import { inject, injectable } from 'inversify';
+import * as lodash from 'lodash';
 import { ReportDiskWriter } from '../report/report-disk-writer';
 import { ReportGenerator } from '../report/report-generator';
+import { ConsoleSummaryReportGenerator } from '../report/summary-report/console-summary-report-generator';
+import { JsonSummaryReportGenerator } from '../report/summary-report/json-summary-report-generator';
+import { SummaryReportData } from '../report/summary-report/summary-report-data';
 import { AIScanner } from '../scanner/ai-scanner';
 import { AxeScanResults } from '../scanner/axe-scan-results';
 import { ScanArguments } from '../scanner/scan-arguments';
 import { CommandRunner } from './command-runner';
 
 @injectable()
 export class FileCommandRunner implements CommandRunner {
+    // tslint:disable-next-line: no-object-literal-type-assertion
+    private readonly summaryReportData = {
+        violationCountByRuleMap: {},
+        urlToReportMap: {},
+    } as SummaryReportData;
+
+    private readonly uniqueUrls = new Set();
+
     constructor(
         @inject(AIScanner) private readonly scanner: AIScanner,
         @inject(ReportGenerator) private readonly reportGenerator: ReportGenerator,
         @inject(ReportDiskWriter) private readonly reportDiskWriter: ReportDiskWriter,
+        @inject(JsonSummaryReportGenerator) private readonly jsonSummaryReportGenerator: JsonSummaryReportGenerator,
+        @inject(ConsoleSummaryReportGenerator) private readonly consoleSummaryReportGenerator: ConsoleSummaryReportGenerator,
+        private readonly fileSystemObj: typeof fs = fs,
+        private readonly lodashObj: typeof lodash = lodash,
     ) {}
 
     public async runCommand(scanArguments: ScanArguments): Promise<void> {
-        const spinner = new Spinner('Running scanner...');
-        let axeResults: AxeScanResults;
+        const spinner = new Spinner(`Running scanner...\n`);
+        spinner.start();
+        // tslint:disable-next-line: no-any
+        const promises: any[] = [];
 
         try {
-            spinner.start();
+            const lines = this.fileSystemObj.readFileSync(scanArguments.inputFile, 'UTF-8').split(/\r?\n/);
 
-            axeResults = await this.scanner.scan(scanArguments.url);
+            lines.forEach(async (line) => {
+                // tslint:disable-next-line: no-parameter-reassignment
+                line = line.trim();
+                if (!this.lodashObj.isEmpty(line) && !this.uniqueUrls.has(line)) {
+                    this.uniqueUrls.add(line);
+                    promises.push(
+                        this.scanURL(line).then((reportContent) => {
+                            const reportName = this.reportDiskWriter.writeToDirectory(scanArguments.output, line, 'html', reportContent);
```

Comment:
even though you are trying to run this in parallel, it will be running in serial, since writeToDirectory is sync. can you change that?

Reply:
I'm trying to scan in parallel since scan is a promise, also each writeTo Disk should be after its scan

Reply:
if the code inside the async function is running synchronously, then trying to run multiple invocations in parallel wont execute them in paralle. You can test it yourselves by doing some logging.

Reply:
again the scan used to be async so the code will be async, if you test (before the last commit), you can see that a new scan used to start before the first finishes